apps/arm/src/main.cpp

```diff
@@ -1,239 +1,158 @@
+      break;
     case HWBRIDGE::CANID::SET_ELBOW_MOTIONDATA:
-      return elbowActuator.setMotionData(motionData);
-    case HWBRIDGE::CANID::SET_WRIST_PITCH_MOTIONDATA:
-      return wristController.setPitchMotionData(motionData);
-    case HWBRIDGE::CANID::SET_WRIST_ROLL_MOTIONDATA:
-      return wristController.setRollMotionData(motionData);
+      Elbow::manager.getActiveController()->setSetPoint(motionData);
+      break;
+    case HWBRIDGE::CANID::SET_LEFT_WRIST_MOTIONDATA:
+      Wrist::leftManager.getActiveController()->setSetPoint(motionData);
+      break;
+    case HWBRIDGE::CANID::SET_RIGHT_WRIST_MOTIONDATA:
+      Wrist::rightManager.getActiveController()->setSetPoint(motionData);
+      break;
     case HWBRIDGE::CANID::SET_CLAW_MOTIONDATA:
-      return clawController.setMotionData(motionData);
+      Claw::manager.getActiveController()->setSetPoint(motionData);
+      break;
+    case HWBRIDGE::CANID::SET_TOOL_TIP_MOTIONDATA:
+      Tooltip::clawTooltipServo.setValue(motionData);
+      break;
     default:
       return MBED_ERROR_INVALID_ARGUMENT;
   }
-
-  return MBED_SUCCESS;
-}
-
-// Run wrist calibration routine
-static mbed_error_status_t runWristCalibration(CANMsg &msg) {
-  bool runCalibration;
-  msg.getPayload(runCalibration);
-
-  if (runCalibration) {
-    return wristController.runPositionCalibration();
-  }
-
   return MBED_SUCCESS;
 }
 
-// Run claw calibration routine
-static mbed_error_status_t runClawCalibration(CANMsg &msg) {
-  bool runCalibration;
-  msg.getPayload(runCalibration);
-
-  if (runCalibration) {
-    return clawController.runPositionCalibration();
+static mbed_error_status_t setPIDParameter(CANMsg &msg) {
+  HWBRIDGE::CONTROL::PID::TuningApiPayload data;
+  msg.getPayload(data);
+
+  const LookupTable::LookupTable<HWBRIDGE::ARM::ActuatorID, Controller::ActuatorControllerManager *> lut = {
+      {HWBRIDGE::ARM::ActuatorID::TURNTABLE, &Turntable::manager},
+      {HWBRIDGE::ARM::ActuatorID::SHOULDER, &Shoulder::manager},
+      {HWBRIDGE::ARM::ActuatorID::ELBOW, &Elbow::manager},
+      {HWBRIDGE::ARM::ActuatorID::WRISTLEFT, &Wrist::leftManager},
+      {HWBRIDGE::ARM::ActuatorID::WRISTRIGHT, &Wrist::rightManager},
+      {HWBRIDGE::ARM::ActuatorID::CLAW, &Claw::manager}};
+
+  auto act = lut.at(data.actuatorID).value_or(nullptr);
+  if (!act) {
+    return MBED_ERROR_INVALID_ARGUMENT;
   }
+  bool success;
+  PID::PID *pid = nullptr;
 
-  return MBED_SUCCESS;
-}
-
-// Deploy or retract tool tip
-static mbed_error_status_t setToolTipDeployment(CANMsg &msg) {
-  bool extend;
-  msg.getPayload(extend);
-
-  if (extend) {
-    return clawController.extendToolTip();
-  } else {
-    return clawController.retractToolTip();
+  switch (msg.getID()) {
+    case HWBRIDGE::CANID::SET_JOINT_PID_DEADZONE:
+      pid     = act->getActiveController()->getPID().value_or(nullptr);
+      success = pid != nullptr;
+      pid->updateDeadzone(data.value);
```

Comment:
Should only do `pid->updateDeadzone()` if `pid` is not a nullptr, like wrap the updateDeadzone() in an if block

Reply:
Same with the below cases

Reply:
Good catch! I've since changed this to return a reference and not a ptr so i think this is fixed.